src/DotnetAffected.Core/ProjectFactory.cs

```diff
@@ -42,8 +42,9 @@ public Project CreateProject(string projectRootElementFilePath)
             var loaded = ProjectCollection.GetLoadedProjects(projectRootElementFilePath);
             if (loaded.Any())
                 return loaded.Single();
-                
+
             var projectRootElement = CreateProjectRootElement(projectRootElementFilePath);
+            // TODO
```

Comment:
When running these tests locally I more or less always (but I have also seen it succeed I think) get this error for the `net8.0` target. All other targets pass just fine:

```
Microsoft.Build.Exceptions.InvalidProjectFileException
A numeric comparison was attempted on "$(_TargetFrameworkVersionWithoutV)" that evaluates to "" instead of a number, in condition "'$(PublishRelease)' == '' and '$(_TargetFrameworkVersionWithoutV)' >= '8.0'".  /usr/local/share/dotnet/sdk/8.0.100/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.DefaultItems.targets
   at Microsoft.Build.Shared.ProjectErrorUtilities.ThrowInvalidProject(String errorSubCategoryResourceName, IElementLocation elementLocation, String resourceName, Object[] args)
   at Microsoft.Build.Shared.ProjectErrorUtilities.ThrowInvalidProject[T1,T2,T3](IElementLocation elementLocation, String resourceName, T1 arg0, T2 arg1, T3 arg2)
   at Microsoft.Build.Evaluation.NumericComparisonExpressionNode.BoolEvaluate(IConditionEvaluationState state, LoggingContext loggingContext)
   at Microsoft.Build.Evaluation.OperatorExpressionNode.TryBoolEvaluate(IConditionEvaluationState state, Boolean& result, LoggingContext loggingContext)
   at Microsoft.Build.Evaluation.AndExpressionNode.BoolEvaluate(IConditionEvaluationState state, LoggingContext loggingContext)
   at Microsoft.Build.Evaluation.OperatorExpressionNode.TryBoolEvaluate(IConditionEvaluationState state, Boolean& result, LoggingContext loggingContext)
   at Microsoft.Build.Evaluation.GenericExpressionNode.Evaluate(IConditionEvaluationState state, LoggingContext loggingContext)
   at Microsoft.Build.Evaluation.ConditionEvaluator.EvaluateConditionCollectingConditionedProperties[P,I](String condition, ParserOptions options, Expander`2 expander, ExpanderOptions expanderOptions, Dictionary`2 conditionedPropertiesTable, String evaluationDirectory, ElementLocation elementLocation, ILoggingService loggingServices, BuildEventContext buildEventContext, IFileSystem fileSystem, ProjectRootElementCacheBase projectRootElementCache, LoggingContext loggingContext)
   at Microsoft.Build.Evaluation.Evaluator`4.EvaluateConditionCollectingConditionedProperties(ProjectElement element, String condition, ExpanderOptions expanderOptions, ParserOptions parserOptions, ProjectRootElementCacheBase projectRootElementCache)
   at Microsoft.Build.Evaluation.Evaluator`4.EvaluatePropertyGroupElement(ProjectPropertyGroupElement propertyGroupElement)
   at Microsoft.Build.Evaluation.Evaluator`4.PerformDepthFirstPass(ProjectRootElement currentProjectOrImport)
   at Microsoft.Build.Evaluation.Evaluator`4.EvaluateImportElement(String directoryOfImportingFile, ProjectImportElement importElement)
   at Microsoft.Build.Evaluation.Evaluator`4.PerformDepthFirstPass(ProjectRootElement currentProjectOrImport)
   at Microsoft.Build.Evaluation.Evaluator`4.EvaluateImportElement(String directoryOfImportingFile, ProjectImportElement importElement)
   at Microsoft.Build.Evaluation.Evaluator`4.PerformDepthFirstPass(ProjectRootElement currentProjectOrImport)
   at Microsoft.Build.Evaluation.Evaluator`4.EvaluateImportElement(String directoryOfImportingFile, ProjectImportElement importElement)
   at Microsoft.Build.Evaluation.Evaluator`4.PerformDepthFirstPass(ProjectRootElement currentProjectOrImport)
   at Microsoft.Build.Evaluation.Evaluator`4.Evaluate()
   at Microsoft.Build.Evaluation.Evaluator`4.Evaluate(IEvaluatorData`4 data, Project project, ProjectRootElement root, ProjectLoadSettings loadSettings, Int32 maxNodeCount, PropertyDictionary`1 environmentProperties, ILoggingService loggingService, IItemFactory`2 itemFactory, IToolsetProvider toolsetProvider, IDirectoryCacheFactory directoryCacheFactory, ProjectRootElementCacheBase projectRootElementCache, BuildEventContext buildEventContext, ISdkResolverService sdkResolverService, Int32 submissionId, EvaluationContext evaluationContext, Boolean interactive)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.Reevaluate(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.ReevaluateIfNecessary(ILoggingService loggingServiceForEvaluation, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext)
   at Microsoft.Build.Evaluation.Project.ProjectImpl.Initialize(IDictionary`2 globalProperties, String toolsVersion, String subToolsetVersion, ProjectLoadSettings loadSettings, EvaluationContext evaluationContext, Boolean interactive)
   at Microsoft.Build.Evaluation.Project.FromProjectRootElement(ProjectRootElement rootElement, ProjectOptions options)
   at DotnetAffected.Core.ProjectFactory.CreateProject(String projectRootElementFilePath) in /Users/trond/Code/dotnet-affected/src/DotnetAffected.Core/ProjectFactory.cs:line 48
   at DotnetAffected.Core.FileSystem.EagerCachingMsBuildGitFileSystem.CreateProjectAndEagerLoadChildren(String path) in /Users/trond/Code/dotnet-affected/src/DotnetAffected.Core/FileSystem/EagerCachingMsBuildGitFileSystem.cs:line 49
   at DotnetAffected.Core.GitChangesProvider.LoadProjectCore(String directory, String pathToFile, String commitRef, Boolean fallbackToHead) in /Users/trond/Code/dotnet-affected/src/DotnetAffected.Core/GitChangesProvider.cs:line 73
   at DotnetAffected.Core.GitChangesProvider.LoadProject(String directory, String pathToFile, String commitRef, Boolean fallbackToHead) in /Users/trond/Code/dotnet-affected/src/DotnetAffected.Core/GitChangesProvider.cs:line 49
   at DotnetAffected.Core.Processor.AffectedProcessor.FindChangedNugetPackages(AffectedProcessorContext context) in /Users/trond/Code/dotnet-affected/src/DotnetAffected.Core/Processor/AffectedProcessor.cs:line 114
   at DotnetAffected.Core.Processor.AffectedProcessor.DiscoverPackageChanges(AffectedProcessorContext context) in /Users/trond/Code/dotnet-affected/src/DotnetAffected.Core/Processor/AffectedProcessor.cs:line 30
   at DotnetAffected.Core.Processor.AffectedProcessorBase.Process(AffectedProcessorContext context) in /Users/trond/Code/dotnet-affected/src/DotnetAffected.Core/Processor/AffectedProcessorBase.cs:line 33
   at DotnetAffected.Core.AffectedExecutor.Execute() in /Users/trond/Code/dotnet-affected/src/DotnetAffected.Core/AffectedExecutor.cs:line 41
   at DotnetAffected.Core.Tests.BaseDotnetAffectedTest.<.ctor>b__1_0() in /Users/trond/Code/dotnet-affected/test/DotnetAffected.Core.Tests/Utils/BaseDotnetAffectedTest.cs:line 16
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at DotnetAffected.Core.Tests.BaseDotnetAffectedTest.get_AffectedSummary() in /Users/trond/Code/dotnet-affected/test/DotnetAffected.Core.Tests/Utils/BaseDotnetAffectedTest.cs:line 24
   at DotnetAffected.Core.Tests.CentralPackageManagementDetectionNestedTests.When_directory_packages_props_changes_without_dependant_projects_nothing_should_be_affected() in /Users/trond/Code/dotnet-affected/test/DotnetAffected.Core.Tests/CentralPackageManagementDetectionNestedTests.cs:line 137
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
```

I'm not sure why this only fails on `net8.0` and not `net6.0` and `net7.0`. What seems to solve it is to add this to the `ProjectOptions` below:

```cs
new ProjectOptions
{
    GlobalProperties = new Dictionary<string, string>{ {"TargetFramework", "net8.0"} },
    // ... existing ProjectOptions
}
```

The tests for non-`net8.0` frameworks still pass with this in place but I'm not sure this is the correct fix. Any thoughts on this?

Reply:
Hi @tanordheim , as you pointed out they were failing due to missing `TargetFramework`. 
I've added it in the sample project generation in tests and that seems to work